homeassistant/components/sensor/sql.py

```diff
@@ -129,15 +129,17 @@ def update(self):
         finally:
             sess.close()
 
+        if not result.returns_rows or result.rowcount == 0:
+            _LOGGER.warning("%s returned no results", self._query)
+            self._state = None
+            self._attributes = {}
+            return
+
         for res in result:
-            _LOGGER.debug(res.items())
+            _LOGGER.debug("result = %s", res.items())
             data = res[self._column_name]
             self._attributes = {k: str(v) for k, v in res.items()}
```

Comment:
Why would we convert this to strings? What about numbers?

Reply:
res.items() will return various columns, so it's not a scalar to convert to a number. 

This call will resort to __str___ of sqlalchemy to get something human readable.

Reply:
It would be better for number types to remain numbers. Otherwise you can't do checks with it in templates without converting it to a number.

Reply:
Now I understand you initial comment. 

Well casting to str() assured me nothing strange would show up in attributes.